src/Core/TTC.idr

```diff
@@ -152,10 +152,10 @@ export
 TTC Constant where
   toBuf b (I x) = do tag 0; toBuf b x
   toBuf b (BI x) = do tag 1; toBuf b x
-  toBuf b (B8 x) = do tag 2; toBuf b x
-  toBuf b (B16 x) = do tag 3; toBuf b x
-  toBuf b (B32 x) = do tag 4; toBuf b x
-  toBuf b (B64 x) = do tag 5; toBuf b x
+  toBuf b (B8 x) = do tag 2; toBuf b $ cast {to = Int} x
+  toBuf b (B16 x) = do tag 3; toBuf b $ cast {to = Int} x
+  toBuf b (B32 x) = do tag 4; toBuf b $ cast {to = Integer} x
+  toBuf b (B64 x) = do tag 5; toBuf b $ cast {to = Integer} x
```

Comment:
Could we get rid of all of these annotations by declaring `TTC BitsN` implementations?